src/Core/Form/IdentifiableObject/DataProvider/OrderReturnFormDataProvider.php

```diff
@@ -0,0 +1,121 @@
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataProvider;
+
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Query\GetOrderReturnForEditing;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\QueryResult\OrderReturnForEditing;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Provides data for order return edit form
+ */
+class OrderReturnFormDataProvider implements FormDataProviderInterface
+{
+    /**
+     * @var CommandBusInterface
+     */
+    private $queryBus;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var string
+     */
+    private $dateFormat;
+
+    /**
+     * @param CommandBusInterface $queryBus
+     */
+    public function __construct(
+        CommandBusInterface $queryBus,
+        RouterInterface $router,
+        TranslatorInterface $translator,
+        string $dateFormat
+    ) {
+        $this->queryBus = $queryBus;
+        $this->router = $router;
+        $this->translator = $translator;
+        $this->dateFormat = $dateFormat;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData($orderReturnId): array
+    {
+        /** @var OrderReturnForEditing $orderReturnForEditing */
+        $orderReturnForEditing = $this->queryBus->handle(new GetOrderReturnForEditing($orderReturnId));
+
+        return [
+            'question' => $orderReturnForEditing->getQuestion(),
+            'customer_name' => $orderReturnForEditing->getCustomerFullName(),
+            'customer_link' => $this->router->generate('admin_customers_view', [
+                'customerId' => $orderReturnForEditing->getCustomerId(),
+            ]),
+            'order' => $this->buildOrderReturnInformation($orderReturnForEditing),
+            'order_link' => $this->router->generate('admin_orders_view', [
+                'orderId' => $orderReturnForEditing->getOrderId(),
+            ]),
+            'order_return_state' => $orderReturnForEditing->getOrderReturnStateId(),
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefaultData(): array
+    {
+        return [];
+    }
+
+    /**
+     * @param OrderReturnForEditing $orderReturnForEditing
+     *
+     * @return string
+     */
+    private function buildOrderReturnInformation(OrderReturnForEditing $orderReturnForEditing): string
+    {
+        return sprintf(
+            '#%s %s %s',
+            $orderReturnForEditing->getOrderId(),
+            $this->translator->trans('from', [], 'Admin.Global'),
+            $orderReturnForEditing->getOrderDate()->format($this->dateFormat)
+        );
```

Comment:
I think you should choose a dedicated wording with placeholders instead:

```suggestion
        return $this->translator->trans(
            '#%order_id% from %order_date%',
            [
                '%order_id%' => $orderReturnForEditing->getOrderId(),
                '%order_date%' => $orderReturnForEditing->getOrderDate()->format($this->dateFormat),
            ],
            'Admin.Orderscustomers.Feature'
        );
```